### PR TITLE
[refactor](if_function) handle result status of function

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

File: be/src/vec/functions/if.cpp [hunk 1]
```diff
@@ -321,23 +321,25 @@ class FunctionIf : public IFunction {
...
-                                    size_t input_rows_count, Status& status) const {
+    Status execute_for_null_then_else(FunctionContext* context, Block& block,
+                                      const ColumnWithTypeAndName& arg_cond,
+                                      const ColumnWithTypeAndName& arg_then,
+                                      const ColumnWithTypeAndName& arg_else, size_t result,
+                                      size_t input_rows_count, bool& handled) const {
         bool then_is_null = arg_then.column->only_null();
         bool else_is_null = arg_else.column->only_null();
 
+        handled = false;
```
Comment (HappenLee) [inline be/src/vec/functions/if.cpp:332]: seems no need set hanled false here ？ out the function have already set the `false`

Comment (mrhhsg) [reply be/src/vec/functions/if.cpp:332]: I think assigning it a value of false might be safer and will not affect the performance.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**